### PR TITLE
Fix SDL crushes when HMI response with wrong type of method

### PR DESCRIPTION
**Pull request by indlu**
Added type check of the received parameter

### CODE REVIEW

**Comment by indlu** [comment]
@istoilovgithub , @LevchenkoS, @okozlovlux, @dcherniev  please, review.

File: src/components/functional_module/src/plugin_manager.cc [hunk 1]
```diff
@@ -175,14 +175,16 @@ ProcessResult PluginManager::ProcessHMIMessage(
...
+               value["result"]["method"].isString()) {
       msg_method = value["result"]["method"].asCString();
       // Error response from HMI
     } else if (value.isMember("error") && value["error"].isMember("data") &&
-               value["error"]["data"].isMember("method")) {
+               value["error"]["data"].isMember("method") &&
+               value["error"]["data"]["method"].isString()) {
       msg_method = value["error"]["data"]["method"].asCString();
     } else {
       LOG4CXX_WARN(logger_,
```
Comment (dcherniev) [inline src/components/functional_module/src/plugin_manager.cc]: @indlu I would suggest to extend text in this warning this way:  "... 'method' field was not found, or was containing an invalid string."

File: src/components/functional_module/src/plugin_manager.cc [hunk 2]
```diff
@@ -232,14 +234,16 @@ bool PluginManager::IsHMIMessageForPlugin(application_manager::MessagePtr msg) {
...
```
Comment (dcherniev) [inline src/components/functional_module/src/plugin_manager.cc]: @indlu Here is the same case as above.

File: src/components/functional_module/src/plugin_manager.cc (hunk 1 above)
Comment (indlu) [inline src/components/functional_module/src/plugin_manager.cc]: @dcherniev , fixed in a7bb34b

File: src/components/functional_module/src/plugin_manager.cc (hunk 2 above)
Comment (indlu) [inline src/components/functional_module/src/plugin_manager.cc]: @dcherniev , fixed in a7bb34b

**Pull request merged**